### PR TITLE
Nessie: Bump Nessie to 0.8.3 / Rename auth_type to auth-type

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: nessie/src/test/java/org/apache/iceberg/nessie/BaseTestIceberg.java [hunk 1]
```diff
@@ -108,7 +108,7 @@ NessieCatalog initCatalog(String ref) {
     newCatalog.setConf(hadoopConfig);
     newCatalog.initialize("nessie", ImmutableMap.of("ref", ref,
         CatalogProperties.URI, uri,
-        "auth_type", "NONE",
+        "auth-type", "NONE",
```
Comment (kbendick) [inline nessie/src/test/java/org/apache/iceberg/nessie/BaseTestIceberg.java:111]: Is this a constant used in the Nessie project itself? If so, perhaps you might consider a follow up for adding `NessieCatalogProperties` class at some point, to help make them more clear to users looking to adopt Nessie coming from the Iceberg repo itself 👍

**Review by kbendick — APPROVED** [review]
Looks good to me with a small comment for a future consideration for follow up.

File: nessie/src/test/java/org/apache/iceberg/nessie/BaseTestIceberg.java (hunk 1 above)
Comment (nastra) [reply nessie/src/test/java/org/apache/iceberg/nessie/BaseTestIceberg.java:111]: yep will consider this for a future follow up, thanks

**Review by rymurr — APPROVED** [review]
Thanks @nastra

**Pull request merged**